bach/bach/series/series_datetime.py

```diff
@@ -348,11 +357,27 @@ def dtype_to_expression(cls, dialect: Dialect, source_dtype: str, expression: Ex
         if source_dtype not in cls.supported_source_dtypes:
             raise ValueError(f'cannot convert {source_dtype} to {cls.dtype}')
 
-        if expression.is_constant and source_dtype == 'string':
-            return cls._constant_expression_to_dtype_expression(dialect, expression)
+        if source_dtype == 'string':
+            if expression.is_constant:
+                return cls._constant_expression_to_dtype_expression(dialect, expression)
+            if is_athena(dialect) and cls.dtype == 'time':
+                # casting varchar to time is a bit tricky, since the string value might not contain
+                # fractional part of the seconds, this breaks date_parse function
+                expression = Expression.construct(
+                    "COALESCE(try(date_parse({}, '%H:%i:%S.%f')), try(date_parse({}, '%H:%i:%S')))",
+                    expression,
+                    expression,
+                )
+                expression = Expression.construct('to_unixtime({})', expression)
 
         return Expression.construct(f'cast({{}} as {cls.get_db_dtype(dialect)})', expression)
 
+    def astype(self, dtype: Union[str, Type]) -> Series:
+        if dtype != SeriesString.dtype and dtype != str:
```

Comment:
```suggestion
        if get_series_type_from_dtype(dtype) is not SeriesString:
```
SeriesString also has a dtype_alias. that we should treat the same as the dtype. Writing it this way checks both the normal dtype and the aliasses, I think

Reply:
fixed here https://github.com/objectiv/objectiv-analytics/pull/1354/commits/6c71c17aaa8a85eb6c6c832658845ea57a15ba93